### PR TITLE
feat: Support for Multiple Inbound Transports

### LINKED ISSUE #1124: Support for Multiple Inbound Transport
Currently, the framework supports multiple outbound transports to be injected but only one inbound transport. The DIDComm Routers need to support both HTTP and WebSocket inbound transports.

Use case - Edge agents without inbound should communicate using WebSocket with it's Router and using HTTP when sending forward messages.

### PR DESCRIPTION
**Pull request by rolsonquadras**
Closes #1124 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/framework/aries/framework.go [hunk 1]
```diff
@@ -250,19 +251,14 @@ func WithPacker(primary packer.Creator, additionalPackers ...packer.Creator) Opt
...
-
 	return context.New(
 		context.WithOutboundDispatcher(a.outboundDispatcher),
 		context.WithMessengerHandler(a.messenger),
 		context.WithOutboundTransports(a.outboundTransports...),
 		context.WithProtocolServices(a.services...),
 		context.WithLegacyKMS(a.kms),
 		context.WithCrypto(a.crypto),
-		context.WithInboundTransportEndpoint(endPoint),
+		context.WithServiceEndpoint(serviceEndpoint(a)),
```
Comment (sudeshrshetty) [inline pkg/framework/aries/framework.go]: Don't you  also need `context.WithRouterEndpoint(...)` here?

**Review by sudeshrshetty — COMMENTED** [review]
Any plan to include this arg in aries-agent-rest?
(and in bddtests docker fixtures)

File: pkg/framework/aries/framework.go (hunk 1 above)
Comment (rolsonquadras) [reply pkg/framework/aries/framework.go]: It's not required here as only requirement for RouterEndpoint is for Route Service.

**Comment by rolsonquadras** [comment]
> Any plan to include this arg in aries-agent-rest?
> (and in bddtests docker fixtures)

I was planning to add it as part of https://github.com/hyperledger/aries-framework-go/issues/989. But this can be done independently - created an issue https://github.com/hyperledger/aries-framework-go/issues/1163

**Review by sudeshrshetty — APPROVED** [review]

File: test/bdd/agent/agent_sdk_steps.go [hunk 2]
```diff
@@ -107,52 +109,84 @@ func (a *SDKSteps) createEdgeAgent(agentID, scheme, routeOpt string) error {
...
 	)
 
-	if inboundPort == "random" {
-		inboundPort = strconv.Itoa(mustGetRandomPort(portAttempts))
+	scheme := strings.Split(schemes, ",")
+	hosts := strings.Split(inboundHosts, ",")
+	ports := strings.Split(inboundPorts, ",")
+	schemeAddrMap := make(map[string]string)
+
+	for i := 0; i < len(scheme); i++ {
```
Comment (troyronda) [inline test/bdd/agent/agent_sdk_steps.go]: a subfunction might help :)

**Review by troyronda — APPROVED** [review]

**Pull request merged**